### DISCUSSION TITLE
How to identify the root node.

### DISCUSSION THREAD

**Discussion by kevin031060** [ds4dm/ecole#222] [Questions] [answered]
In the example, the StrongBranchingScores() is used with the 0.05 probability. However, I want to execute the SB always at the roor node. Is there any way to realize it? A naive but time-consuming way is to calculate the SB scores all the time and use it to select actions always at the first step. 
More generally, I want to execute different strategies at different stages when solving an instance. It cost a lot time if I compute the SB socres (or execute other heavy models) for each step.  But the env initialization with observation function procedure is at the beginning. Is there any idea to achieve this

**Comment by gasse** [comment, marked as answer]
Hi @kevin031060 ,

You can decide whether to extract or not the SB scores at each time step based on the current state of SCIP. For example, you can check that the currently processed node is the root node as follows:
```python
m = model.as_pyscipopt()
if m.getCurrentNode().getDepth() == 0:
  # do something only at the root node
```

Best,
Maxime

**Reply by kevin031060** [reply]
Wow. Thanks. I'm not familiar with the Pyscipopt APIs. I'll check it.
